internal/acctest: Unit Testing Panic With Endpoint Configuration Reflection

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform AWS Provider Version

N/A and main branch

### Affected Test

* TestAccProvider_endpoints

### Expected Behavior

No panic. 😄 

### Actual Behavior

Panic at the disco:

```
--- FAIL: TestAccProvider_endpoints (10.98s)
panic: reflect: call of reflect.Value.FieldByName on zero Value [recovered]
        panic: reflect: call of reflect.Value.FieldByName on zero Value

goroutine 65 [running]:
testing.tRunner.func1.2({0x858d460, 0xc039bc0b10})
        /usr/local/Cellar/go/1.18/libexec/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
        /usr/local/Cellar/go/1.18/libexec/src/testing/testing.go:1392 +0x39f
panic({0x858d460, 0xc039bc0b10})
        /usr/local/Cellar/go/1.18/libexec/src/runtime/panic.go:838 +0x207
reflect.flag.mustBe(...)
        /usr/local/Cellar/go/1.18/libexec/src/reflect/value.go:223
reflect.Value.FieldByName({0x0?, 0x0?, 0xc0244d7250?}, {0x9bd3d4b?, 0x1?})
        /usr/local/Cellar/go/1.18/libexec/src/reflect/value.go:1297 +0x1d6
github.com/hashicorp/terraform-provider-aws/internal/acctest.testAccCheckEndpoints.func1(0xc01e4f6240?)
        /Users/bflad/src/github.com/hashicorp/terraform-provider-aws/internal/acctest/provider_test.go:855 +0x39a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.ComposeTestCheckFunc.func1(0xa82dd10?)
        /Users/bflad/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/testing.go:724 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.testStepNewConfig({_, _}, {_, _}, {0x0, 0xc0086abfe0, 0xc007209b30, 0x0, 0x0, 0x0, ...}, ...)
        /Users/bflad/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/testing_new_config.go:135 +0xa17
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.runNewTest({0xa82dd10, 0xc01a390d80}, {0xa845608, 0xc0057069c0}, {0x0, 0xc0086abfe0, 0xc007209b30, 0x0, 0x0, 0x0, ...}, ...)
        /Users/bflad/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/testing_new.go:170 +0x853
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.Test({0xa845608, 0xc0057069c0}, {0x0, 0xc0086abfe0, 0xc007209b30, 0x0, 0x0, 0x0, 0x0, 0x0, ...})
        /Users/bflad/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/testing.go:645 +0x46f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.ParallelTest({0xa845608, 0xc0057069c0}, {0x0, 0xc0086abfe0, 0xc007209b30, 0x0, 0x0, 0x0, 0x0, 0x0, ...})
        /Users/bflad/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/resource/testing.go:555 +0x88
github.com/hashicorp/terraform-provider-aws/internal/acctest.TestAccProvider_endpoints(0xc0057069c0)
        /Users/bflad/src/github.com/hashicorp/terraform-provider-aws/internal/acctest/provider_test.go:182 +0x3a7
testing.tRunner(0xc0057069c0, 0x9dc6518)
        /usr/local/Cellar/go/1.18/libexec/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
        /usr/local/Cellar/go/1.18/libexec/src/testing/testing.go:1486 +0x35f
```

### Steps to Reproduce

1. `TF_ACC=1 go test -count=1 -v ./internal/acctest`

### Important Factoids

My local development environment is using Go 1.18, in case this is a new behavior in the `reflect` package.

This unit test was designed to loop through all the possible services and ensure that any configuration provided in the `provider` configuration block `endpoints` block was actually set correctly on the service clients.

I believe the panic is occurring because the `route53domains` service uses aws-sdk-go-v2, which does not have the same struct types and fields for accessing configuration information as v1 of aws-sdk-go. Adding the below to `internal/acctest/provider_test.go` at line 849 works around the issue:

```
if serviceKey == "route53domains" {
	continue
}
```

However, a real fix for this would require reflecting into the newer aws-sdk-go-v2 Client types or otherwise extracting the endpoint configuration from them as appropriate.

--- a/internal/acctest/provider_test.go
+++ b/internal/acctest/provider_test.go
@@ -852,6 +852,10 @@ func testAccCheckEndpoints(providers *[]*schema.Provider) resource.TestCheckFunc
 					return fmt.Errorf("unable to match conns.AWSClient struct field name for endpoint name: %s", serviceKey)
 				}
 
+				if !reflect.Indirect(providerClientField).FieldByName("Config").IsValid() {
+					continue // currently unknown how to do this check for v2 clients
+				}
+
 				actualEndpoint := reflect.Indirect(reflect.Indirect(providerClientField).FieldByName("Config").FieldByName("Endpoint")).String()
 				expectedEndpoint := fmt.Sprintf("http://%s", serviceKey)
 

